src/Microsoft.AspNetCore.Mvc.Core/RequireHttpsAttribute.cs

```diff
@@ -82,6 +88,9 @@ protected virtual void HandleNonHttpsRequest(AuthorizationFilterContext filterCo
                     host = new HostString(host.Host);
                 }
 
+                //i use MvcOption.requireHttpsPermanent value if Permanent parameter is null
+                _permanent = _permanent ?? optionsAccessor.Value.RequireHttpsPermanent;
```

Comment:
i think the options don't change, i can use a new field but it will have the value from optionsAccessor.Value.RequireHttpsPermanent for each call...